actor/src/main/scala/org/apache/pekko/actor/ActorCell.scala

```diff
@@ -643,7 +643,7 @@ private[pekko] class ActorCell(
     def failActor(): Unit =
       if (_actor != null) {
         clearActorFields(actor, recreate = false)
-        setFailedFatally()
+        setFailed(actor.self)
```

Comment:
I am not sure if this change will break other cases or not, I will investigate later.

relate PR:

- https://github.com/akka/akka/pull/29098/files
- https://github.com/akka/akka/pull/27614/files

Reply:
After investigation, we can not set the fault state here... otherwise, we can distinguish between fatal and normal fault

Reply:
(these links to Akka PRs are OK because they're old, pre-BSL PRs :) )